core/dbt/include/global_project/macros/materializations/tests/helpers.sql

```diff
@@ -3,12 +3,27 @@
 {%- endmacro %}
 
 {% macro default__get_test_sql(main_sql, fail_calc, warn_if, error_if, limit) -%}
+    {% set model_name = model.file_key_name.split('.')[1] %}
+    {% set warn_if, warn_pct_division = get_pct_division(warn_if, model_name) %}
+    {% set error_if, error_pct_division = get_pct_division(error_if, model_name) %}
+
     select
       {{ fail_calc }} as failures,
-      {{ fail_calc }} {{ warn_if }} as should_warn,
-      {{ fail_calc }} {{ error_if }} as should_error
+      {{ fail_calc }} {{ warn_pct_division }} {{ warn_if }} as should_warn,
+      {{ fail_calc }} {{ error_pct_division }} {{ error_if }} as should_error
     from (
       {{ main_sql }}
       {{ "limit " ~ limit if limit != none }}
     ) dbt_internal_test
 {%- endmacro %}
+
+
+{% macro get_pct_division(threshold, model_name) %}
+  {% if threshold[-1] == '%' %}
```

Comment:
Could you imagine someone wanting to define a `%` failure threshold for a singular test (as opposed to generic tests)? I can't think of one myself

Reply:
We took a list at that this morning, I think I probably could think of someone wanting to do this (in a rare case), but it wasn't clear to me why the current approach wouldn't work (given that thresholds should work in the same way).  Could you help me understand how that might work?

Reply:
In the case where a singular test depends on multiple nodes, I'm not sure how we can tell which node should represent the denominator?

I also think, in a singular test, you're writing all the SQL yourself, it's pretty trivial to add a `/ (select count(*) from total * 100` and define `warn_if`/`error_if` with percents-as-integers.